core/templates/dev/head/exploration_editor/EditorServices.js

```diff
@@ -1952,3 +2012,283 @@ oppia.factory('explorationWarningsService', [
+                lostChangeValue = angular.element(
+                  '<div>Edited answer group: </div>');
+                if (newValue.outcome.dest !== oldValue.outcome.dest) {
+                  changesEl.append(
+                    '<p>Destination: ' + newValue.outcome.dest + '</p>');
+                }
+                if (!angular.equals(
+                  newValue.outcome.feedback, oldValue.outcome.feedback)) {
+                  changesEl.append(
+                    '<p>Feedback: ' + newValue.outcome.feedback + '</p>');
+                }
+                if (!angular.equals
+                    (newValue.rule_specs, oldValue.rule_specs)) {
+                  var rulesList = getRulesList(newValue);
+                  if (rulesList.length) {
+                    changesEl.append('<div>Rules: <div>');
+                    for (var rule in rulesList) {
+                      changesEl.append(rulesList[rule]);
+                    }
+                  }
+                }
+                lostChangeValue.append(changesEl);
+              } else {
+                lostChangeValue = 'Deleted answer group';
+              }
+              break;
+
+            case 'default_outcome':
+              var defaultOutcomeChanges = getRelativeChangeToGroups(lostChange);
+              var changesEl = angular.element('<div></div>');
+              if (defaultOutcomeChanges === 'added') {
+                lostChangeValue = angular.element(
+                  '<div>Added default outcome: </div>');
+                changesEl.append(
+                  '<p>Destination: ' + newValue.dest + '</p>');
+                changesEl.append(
+                  '<p>Feedback: ' + newValue.feedback + '</p>');
+                lostChangeValue.append(changesEl);
+              } else if (defaultOutcomeChanges === 'edited') {
+                lostChangeValue = angular.element(
+                  '<div>Edited default outcome: </div>');
+                if (newValue.dest !== oldValue.dest) {
+                  changesEl.append(
+                    '<p>Destination: ' + newValue.dest + '</p>');
+                }
+                if (!angular.equals(
+                  newValue.feedback, oldValue.feedback)) {
+                  changesEl.append(
+                    '<p>Feedback: ' + newValue.feedback + '</p>');
+                }
+                lostChangeValue.append(changesEl);
+              } else {
+                lostChangeValue = 'Deleted default outcome';
+              }
+              break;
+          };
+
+          var existingEditsToState = statesWiseEditsMapping[stateName];
+          if (!existingEditsToState) {
+            statesWiseEditsMapping[stateName] = [];
+          }
+          statesWiseEditsMapping[stateName].push(lostChangeEl);
+          break;
+      }
+
+      if (lostChangeValue) {
+        lostChangeEl.html(lostChangeValue);
+        if (lostChangeEl) {
+          outerHtml.append(lostChangeEl);
```

Comment:
This is strange to me. I would have thought you'd only want to add to outerHtml here for commands other than EDIT_STATE_PROPERTY. I think it would be clearer if for the other commands you directly append to outerHtml, and then get rid of this variable entirely. The only long-term variables you need should be outerHtml and then statesWiseEditsMapping for the edits within states.